q2_types/feature_data/_transformer.py

```diff
@@ -408,3 +408,96 @@ def _224(data: pd.DataFrame) -> DifferentialFormat:
     ff = DifferentialFormat()
     qiime2.Metadata(data).save(str(ff))
     return ff
+
+
+# Protein FASTA transformers
+
+class ProteinIterator(collections.abc.Iterable):
+    def __init__(self, generator):
+        self.generator = generator
+
+    def __iter__(self):
+        yield from self.generator
+
+
+class AlignedProteinIterator(ProteinIterator):
+    pass
```

Comment:
Do you think we need these new iterators? Or can we use the old iterators, and just make transformers from Protein formats to/from the existing iterator types?

Reply:
I don't think we need those _per se_. It would probably be much simpler to have a more generic ones like `FASTAIterator`, `AlignedFASTAIterator` (and `PairedFASTAIterator` ??) and use those, regardless of the data type. 
At the same time, I guess we cannot just rename the DNAIterators as that would probably break things in a lot of other plugins. That is why I introduced those new ones - to be more explicit about the data they are operating on, even though the iterators look exactly the same.
So, if you think we should re-use the DNA-ones - that would work, I just felt it would be a bit confusing...